libV2/config/index.js

```diff
@@ -297,8 +297,27 @@ class Config {
 
         const diskUsage = {
             path: _loadFromEnv('DISK_USAGE_PATH', (config.diskUsage || {}).path),
+            mode: _loadFromEnv('DISK_USAGE_MODE', (config.diskUsage || {}).mode),
+            softLimit: _loadFromEnv('DISK_USAGE_SOFT_LIMIT', (config.diskUsage || {}).softLimit, _typeCasts.diskSize),
+            hardLimit: _loadFromEnv('DISK_USAGE_HARD_LIMIT', (config.diskUsage || {}).hardLimit, _typeCasts.diskSize),
+            expirationBlockSize: _loadFromEnv(
+                'DISK_USAGE_EXPIRATION_BLOCK_SIZE',
+                (config.diskUsage || {}).expirationBlockSize,
+                _typeCasts.int,
+            ),
         };
 
+
+        if (!diskUsage.path && (diskUsage.softLimit !== undefined || diskUsage.hardLimit !== undefined)) {
+            throw Error('You must specify diskUsage.path to monitor for disk usage');
+        } else if (diskUsage.path && (diskUsage.softLimit === undefined && diskUsage.hardLimit === undefined)) {
+            throw Error('One of diskUsage.softLimit or diskUsage.hardLimit must be specified');
+        }
```

Comment:
I think the if checks here are a little redundant if I am understanding it right. Is there a reason why this wouldn't work?

```suggestion
        if (!diskUsage.path) {
            throw Error('You must specify diskUsage.path to monitor for disk usage');
        }
        
        if (diskUsage.softLimit === undefined && diskUsage.hardLimit === undefined) {
            throw Error('One of diskUsage.softLimit or diskUsage.hardLimit must be specified');
        }
```

Reply:
The logic is meant to be, if you define `path` then you must define `softLimit` or `hardLimit`and conversely if you define `softLimit` or `hardLimit` you must define `path`.
Defining nothing is also an option as this config file is shared amonst the various utapi components which might not need the `duskUsage` section

Reply:
Also we need this checking here as config loading can happen completely though the environment

Reply:
oooh ok, that is more complex than I was thinking. It looks good then, that matches.